addons/mail/static/src/models/message/message.js

```diff
@@ -564,6 +575,91 @@ function factory(dependencies) {
             return [['replace', threads]];
         }
 
+        /**
+         * Modifies the message to add the 'read more/read less' functionality
+         * All element nodes with 'data-o-mail-quote' attribute are concerned.
+         * All text nodes after a ``#stopSpelling`` element are concerned.
+         * Those text nodes need to be wrapped in a span (toggle functionality).
+         * All consecutive elements are joined in one 'read more/read less'.
+         *
+         * FIXME This method should be rewritten (task-2308951)
+         *
+         * @private
+         * @param {jQuery} $element
+         */
+         _insertReadMoreLess($element) {
```

Comment:
This doesn't appear to be using `this` except for reading the `env` so I would at least make it static.

Reply:
Or actually what's the point of this method instead of directly doing everything in `_computePrettyBodyWithReadMoreLess`?
Looks like a useless indirection actually.

Reply:
Essentially a move from component to model of this code. Also the function is recursive.